scripts/diagnostics/vulkan/stage_cmp.cpp

```
@@ -0,0 +1,105 @@
+    if (const char* ov = getenv("STAGE_CMP_EMBEDS")) {
+        FILE* g = fopen(ov, "rb");
+        fseek(g, 0, SEEK_END); long gs = ftell(g); fseek(g, 0, SEEK_SET);
+        ie.data.resize(gs / 4);
+        fread(ie.data.data(), 4, gs / 4, g); fclose(g);
+        ie.n_tokens = (int64_t)prompt.ids.size(); ie.width = m.config.llm.hidden;
+        stat("inputs_embeds(OVR)", ie.data);
+    } else {
+        if (!moss::build_inputs_embeds(m, prompt, adap, ie, e)) { fprintf(stderr, "embeds: %s\n", e.c_str()); return 2; }
+        stat("inputs_embeds", ie.data);
+        FILE* d = fopen("/tmp/embeds_dump.f32", "wb");
+        fwrite(ie.data.data(), 4, ie.data.size(), d); fclose(d);
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)